packages/headless/src/features/facets/category-facet-set/category-facet-utils.ts

```diff
@@ -33,16 +33,46 @@ export function partitionIntoParentsAndValues<
   return {parents, values};
 }
 
-export function getActiveValueFromValueTree(
-  valuesAsTrees: CategoryFacetValue[]
-): CategoryFacetValue | undefined {
+export function findActiveValueAncestry<T extends GenericCategoryFacetValue>(
+  valuesAsTree: T[]
+): T[] {
+  const ancestryMap = new Map<T, T>();
+  const activeValue = getActiveValueFromValueTree(valuesAsTree, ancestryMap);
+  return activeValue ? getActiveValueAncestry(activeValue, ancestryMap) : [];
+}
+
+function getActiveValueFromValueTree<TValue extends GenericCategoryFacetValue>(
+  valuesAsTrees: TValue[],
+  ancestryMap?: Map<TValue, TValue>
+): TValue | undefined {
   const valueToVisit = [...valuesAsTrees];
   while (valueToVisit.length > 0) {
     const currentValue = valueToVisit.shift()!;
     if (currentValue.state === 'selected') {
       return currentValue;
     }
-    valueToVisit.unshift(...currentValue.children);
+    if (ancestryMap) {
+      for (const childValue of currentValue.children) {
+        ancestryMap.set(childValue as TValue, currentValue);
+      }
+    }
+    valueToVisit.unshift(...(currentValue.children as TValue[]));
   }
   return undefined;
 }
+
+function getActiveValueAncestry<TValue extends GenericCategoryFacetValue>(
+  activeValue: TValue | undefined,
+  valueToParentMap: Map<TValue, TValue>
+): TValue[] {
+  const activeValueAncestry: TValue[] = [];
+  if (!activeValue) {
+    return [];
+  }
+  let lastParent: TValue | undefined = activeValue;
```

Comment:
`undefined` because Map.get returns undefined if the key ain't in the map.
This is used in the while as our cue to break the loop